src/rust/engine/rule_graph/src/lib.rs

```diff
@@ -512,19 +513,16 @@ impl<R: Rule> RuleGraph<R> {
 ///
 #[derive(Eq, PartialEq, Clone, Debug)]
 pub struct RuleEdges<R: Rule> {
-  dependencies: HashMap<R::DependencyKey, Vec<Entry<R>>>,
```

Comment:
For my own edification, since I don't know this code - how does an `Intern<Entry<R>>` (which presumably represents a single entry) replace a `Vec<Entry<R>>` (which appears to be able to represent a list of them)? Was that `Vec` always a singleton, but had to exist as a referenceable container or something?

Reply:
Good catch. This was a semantic change: see the TODO immediately above which I forgot to remove!